runtime/runtime/src/ext.rs

```diff
@@ -378,13 +377,12 @@ impl<'a> Contract for RuntimeContractExt<'a> {
         let account_id = self.account_id;
         let code_hash = self.hash();
         let version = self.current_protocol_version;
-        let chain_id = self.chain_id;
         if checked_feature!("stable", EthImplicitAccounts, version)
             && account_id.get_account_type() == AccountType::EthImplicitAccount
-            && code_hash_matches_wallet_contract(chain_id, &code_hash, version)
         {
-            return Some(wallet_contract(&chain_id, version));
+            return wallet_contract(code_hash);
```

Comment:
I don't think this early return is quite right because of the accounts that existed prior to the eth-implicit accounts protocol change (these accounts are discussed in the description of [this PR](https://github.com/near/nearcore/pull/11606)). We should only return the wallet contract when the code hash matches one of the wallet contracts (i.e. `wallet_contract` returns `Some`) and otherwise we should proceed passed this `if` block to the normal account operation.

Reply:
Addressed.